src/gui/plugins/component_inspector/Light.qml

```diff
@@ -0,0 +1,1083 @@
+      directionXItem.value,
+      directionYItem.value,
+      directionZItem.value,
+      innerAngleItem.value,
+      outerAngleItem.value,
+      falloffItem.value,
+      model.data[19]
+    );
+  }
+
+  FontMetrics {
+    id: fontMetrics
+    font.family: "Roboto"
+  }
+
+  /**
+   * Used to create a spin box
+   */
+  Component {
+    id: spinZeroOne
+    IgnSpinBox {
+      id: writableSpin
+      value: writableSpin.activeFocus ? writableSpin.value : numberValue
+      minimumValue: 0
+      maximumValue: 1
+      decimals: 6
+      onEditingFinished: {
+        sendLight()
+      }
+    }
+  }
+  Component {
+    id: spinZeroMax
+    IgnSpinBox {
+      id: writableSpin
+      value: writableSpin.activeFocus ? writableSpin.value : numberValue
+      minimumValue: 0
+      maximumValue: 1000000
+      decimals: 6
+      onEditingFinished: {
+        sendLight()
+      }
+    }
+  }
+  Component {
+    id: spinNoLimit
+    IgnSpinBox {
+      id: writableSpin
+      value: writableSpin.activeFocus ? writableSpin.value : numberValue
+      minimumValue: -100000
+      maximumValue: 100000
+      decimals: 6
+      onEditingFinished: {
+        sendLight()
+      }
+    }
+  }
+
+  Component {
+    id: readOnlyNumber
+    Text {
+      id: numberText
+      anchors.fill: parent
+      horizontalAlignment: Text.AlignRight
+      verticalAlignment: Text.AlignVCenter
+      text: {
+        var decimals = numberText.width < 100 ? 2 : 6
+        return numberValue.toFixed(decimals)
+      }
+    }
+  }
+
+  Column {
+    anchors.fill: parent
+
+    // Header
+    Rectangle {
+      id: header
+      width: parent.width
+      height: typeHeader.height
+      color: "transparent"
+
+      RowLayout {
+        anchors.fill: parent
+        Item {
+          width: margin
+        }
+        Image {
+          id: icon
+          sourceSize.height: indentation
+          sourceSize.width: indentation
+          fillMode: Image.Pad
+          Layout.alignment : Qt.AlignVCenter
+          source: content.show ?
+              "qrc:/Gazebo/images/minus.png" : "qrc:/Gazebo/images/plus.png"
+        }
+        TypeHeader {
+          id: typeHeader
+        }
+        Item {
+          Layout.fillWidth: true
+        }
+      }
+      MouseArea {
+        anchors.fill: parent
+        hoverEnabled: true
+        cursorShape: Qt.PointingHandCursor
+        onClicked: {
+          content.show = !content.show
+        }
+        onEntered: {
+          header.color = highlightColor
+        }
+        onExited: {
+          header.color = "transparent"
+        }
+      }
+    }
+
+    // Content
+    Rectangle {
+      id: content
+      property bool show: false
+      width: parent.width
+      height: show ? grid.height : 0
+      clip: true
+      color: "transparent"
+
+      Behavior on height {
+        NumberAnimation {
+          duration: 200;
+          easing.type: Easing.InOutQuad
+        }
+      }
+
+
+      GridLayout {
```

Comment:
I don't think a grid layout works well for this widget. See how the RGBA spin boxes are squished in a tiny cell, leaving a lot of empty space on their left, instead of expanding to use as much space as possible.

![image](https://user-images.githubusercontent.com/5751272/103047061-23c06680-453f-11eb-9a4c-bc2fadc6b46b.png)

You can try increasing their `columnSpan` so they occupy that empty space. I recommend you draw the layout in a grid and see how many cells you expect each widget to occupy.

Another option is using a `ColumnLayout` with multiple other layouts inside of it, instead of a large `GridLayout`. This would give each item in the column more freedom, because their columns don't need to match. For example, the `Cast shadows` text is large and stretches that whole column right now. If you had separate layouts, that row wouldn't influence the others.

Also, another thing to consider while working on the layout would be to use sliders instead of spins for the 0-1 bounded values. Or maybe slider + spin.

Annnd another cool thing to do for colors is to use a `ColorDialog` like the `GridConfig` plugin does.